File: frontend/src/types/analysis.ts
export interface CollegeOption {
  college_code: string;
  college_name: string;
  city?: string;
  district?: string;
}

export interface CourseOption {
  course_code: string;
  course_name: string;
}

export interface FilterOptions {
  years: number[];
  rounds: number[];
  categories: string[];
  seat_sections: string[];
  stages: string[];
  cities_districts: string[];
  colleges: CollegeOption[];
  courses: CourseOption[];
}

export interface CutoffSummary {
  total_matches: number;
  min_percentile?: number;
  max_percentile?: number;
  avg_percentile?: number;
  min_merit?: number;
  max_merit?: number;
  unique_colleges: number;
  unique_courses: number;
}

export interface RoundComparisonItem {
  course_code: string;
  course_name: string;
  college_name: string;
  category_code: string;
  seat_section: string;
  stage: string;
  round_a_merit?: number;
  round_a_percentile?: number;
  round_b_merit?: number;
  round_b_percentile?: number;
  percentile_change?: number;
}

export interface RoundComparison {
  course_id: number;
  course_name: string;
  college_name: string;
  category: string;
  round_1_score?: number;
  round_2_score?: number;
  round_3_score?: number;
  diff_1_2?: number;
  diff_2_3?: number;
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Dict, Any

class RoundComparisonItem(BaseModel):
    course_code: str
    course_name: str
    college_name: str
    category_code: str
    seat_section: str
    stage: str
    round_a_merit: Optional[int] = None
    round_a_percentile: Optional[float] = None
    round_b_merit: Optional[int] = None
    round_b_percentile: Optional[float] = None
    percentile_change: Optional[float] = None

class DataQualityReport(BaseModel):
    total_records: int
    records_with_missing_merit: int
    records_with_missing_percentile: int
    records_with_zero_merit: int
    duplicate_count: int
    records_by_validation_status: Dict[str, int]
    category_distribution: Dict[str, int]
    section_distribution: Dict[str, int]

class CategorySummary(BaseModel):
    category_code: str
    count: int

class CollegeOption(BaseModel):
    college_code: str
    college_name: str
    city: Optional[str] = None
    district: Optional[str] = None
    college_type: Optional[str] = None
    funding_type: Optional[str] = None

class CourseOption(BaseModel):
    course_code: str
    course_name: str

class FilterOptions(BaseModel):
    years: List[int]
    rounds: List[int]
    categories: List[str]
    seat_sections: List[str]
    stages: List[str]
    cities_districts: List[str]
    colleges: List[CollegeOption]
    courses: List[CourseOption]

class CutoffSummary(BaseModel):
    total_matches: int
    min_percentile: Optional[float] = None
    max_percentile: Optional[float] = None
    avg_percentile: Optional[float] = None
    min_merit: Optional[int] = None
    max_merit: Optional[int] = None
    unique_colleges: int = 0
    unique_courses: int = 0

